src/test/scripts/check_enterprise_merge.sh

```diff
@@ -4,52 +4,55 @@
-try_merge() {
-    to=$1
-    from=$2
-    git checkout "${to}"
-    # this will exit since -e option is set and it will return non-zero code on conflicts.
-    git merge --no-ff --no-commit "${from}"
-
-}
+PR_BRANCH="${CIRCLE_BRANCH}"
 
-cd /tmp
-if [ ! -d citus-enterprise ]; then
-    git clone https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise
-fi
+git remote add enterprise "https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise"
+# Prevent any pushes
+git remote set-url --push origin no-pushing
+git remote set-url --push enterprise no-pushing
 
-cd citus-enterprise
 git config user.email "citus-bot@microsoft.com"
 git config user.name "citus bot"
 
-# reset repository into usable state if script ran before
-git fetch origin
-git reset --hard
-git checkout enterprise-master
-git reset --hard origin/enterprise-master
-
-branch_name="${CIRCLE_BRANCH}"
-
-# check if the branch on community exists on enterprise
-# the output will not be empty if it does
-if [ `git branch -r --list origin/$branch_name` ]
-then
-    try_merge enterprise-master origin/$branch_name
-else
-    # add community as a remote if not already added
-    set +e
-    if ! git ls-remote community > /dev/null; then
-        set -e
-        git remote add --no-tags community git@github.com:citusdata/citus.git
-    fi
-    set -e
-
-    # prevent pushes to community and update branch we care about
-    git remote set-url --push community no-pushing
-    git fetch community $branch_name
-
-    try_merge enterprise-master community/$branch_name
+# Fetch enterprise-master
+git fetch enterprise enterprise-master
+
+
+# Try to fetch the enterprise version of the branch. If it fails we assume it
+# does not exist. This seems reasonble, because even in case of other failures
+# (e.g. network) we still continue as if the enterprise version of the branch
+# does not exist.
+if ! git fetch enterprise "$PR_BRANCH" > /dev/null 2>&1 ; then
+    echo "INFO: enterprise/$PR_BRANCH was not found"
+    # If the current branch does not exist on the enterprise repo, then all we
+    # have to check is if it can be merged into enterprise master without
+    # problems.
+    # this will exit since -e option is set and it will return non-zero code on conflicts.
+    git checkout enterprise-master
+    # Check if we can merge the PR branch into enterprise-master
+    git merge --no-ff --no-commit "origin/$PR_BRANCH"
+    exit 0
 fi
+
+# Show the top commit of the enterprise branch to make debugging easier
+git log -n 1 "enterprise/$PR_BRANCH"
+
+# Check that this branch contains both the top commits of community PR branch
+# and the top commit of enterprise-master. This means this branch has them
+# merged.
+
+# The "2> /dev/null" bit is to make sure that the "set -x" output is ignored
+# for the subshell
+git merge-base --is-ancestor enterprise/enterprise-master "enterprise/$PR_BRANCH" \
```

Comment:
I find this block a bit hard to read

Reply:
I've made this regular `if` statements now.